packages/backend/src/api/request.ts

```diff
@@ -91,12 +91,17 @@ export function buildRequest(options: BuildRequestOptions) {
 
     // Build headers
     const headers: Record<string, any> = {
-      Authorization: `Bearer ${secretKey}`,
       'Clerk-API-Version': SUPPORTED_BAPI_VERSION,
       'User-Agent': userAgent,
       ...headerParams,
     };
 
+    if (secretKey) {
+      Object.assign(headers, {
+        Authorization: `Bearer ${secretKey}`,
+      });
+    }
```

Comment:
Do not fear mutation... 😇

```suggestion
    if (secretKey) {
      headers.Authorization = `Bearer ${secretKey}`,
    }
```

Alternatively, move into headers declaration:
```ts
{
  ...(secretKey && { Authorization: 'Bearer ${secretKey}` })
}